src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -266,7 +266,13 @@ private void RunAssembly(string assemblyName, TestFilter filter)
                     }
                 }
                 else
-                    TestLog.Info("NUnit failed to load " + assemblyName);
+                {
+                    var msgNode = loadResult.SelectSingleNode("properties/property[@name='_SKIPREASON']");
+                    if (msgNode != null && (new[] { "contains no tests", "Has no TestFixtures" }).Any(msgNode.GetAttribute("value").Contains))
+                        TestLog.Info("NUnit couldn't find any tests in " + assemblyName + ". (Probably an assembly dependency problem in your projects.");
```

Comment:
That's why I had not approved this earlier, although I can't see my comment here!

This is a perennial problem! From the perspective of the user encountering the problem, it seems that most cases of a particular error probably arise from a certain cause, so they want the message to explain the cause. However, such a message would be misleading in many other cases.

Idea: Can we have a link in the message to an extended explanation? Would a link work in the output window?